dashboard/app/models/pd/application/facilitator1819_application.rb

```diff
@@ -439,13 +439,18 @@ def additional_text_fields
     # @override
     # Filter out extraneous answers, based on selected program (course)
     def self.filtered_labels(course)
-      labels_to_remove = (course == 'csf' ?
-        [:csd_csp_fit_availability, :csd_csp_teachercon_availability]
-        : # csd / csp
-        [:csf_availability, :csf_partial_attendance_reason]
-      )
+      # memoize in a hash, per course
+      @@filtered_labels ||= Hash.new do |h, key|
```

Comment:
rather than implementing this as a class variable (which I'm pretty sure are generally discouraged in Rails), could this just be a constant?